File: cpp/n258_AddDigits.cpp
// https://leetcode-cn.com/problems/add-digits/
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;
class Solution {
public:
    int addDigits(int num) {
        int res = num;
        while (res >= 10){
            res = 0;
            while (num > 0){
                res += num % 10;
                num = num / 10;
            }
            num = res;
        }
        return res;
    }
};

int main(){
    return 0;
}

File: python/n258_AddDigits.py
# https://leetcode-cn.com/problems/add-digits/
class Solution:
    def addDigits(self, num: int) -> int:
        res = num
        while res >= 10:
            res = 0
            while num > 0:
                res += num % 10
                num //= 10
            num = res
        return res 


if __name__ == "__main__":
    solu = Solution()
    print(solu.addDigits(38))